metricbeat/module/redis/info/info.go

```diff
@@ -171,20 +177,34 @@ func (m *MetricSeter) Setup(cfg *ucfg.Config) error {
 
 func (m *MetricSeter) Fetch(ms *helper.MetricSet) (events []common.MapStr, err error) {
 	for _, host := range ms.Config.Hosts {
-		c := m.redisPools[host].Get()
-		out, err := rd.String(c.Do("INFO"))
-		c.Close()
-		if err != nil {
-			logp.Err("Error converting to string: %v", err)
-		}
+		// Fetch default INFO
+		info := m.fetchRedisStats(host, "default")
+
+		// TODO: Take this approach for all sub types, or grep out keyspace from info
+		//
+		// Fetch keyspace metrics, so they can be easily iterated over later.
+		// It might be nice to take this approach for all sub stats from
+		// INFO and allow user to configure which sub metric sets they collect.
+		keyspace := m.fetchRedisStats(host, "keyspace")
```

Comment:
My idea was that the info metricset call to Redis always uses "all" but then does the filtering on the code side.

Something like this (not running code):

```
all := m.fetchRedisStats(host, "all")

// Directly "Magically" also parses keyspace
completeEvent := parseRedisInfo("all")

// stats coming from config, can be for example "server,cluster"
event = eventMapping(completeEvent, stats)
```

So my assumption is that keyspace is always fetched and automatically parsed. Then on the config side we can define which parts should be returned with a "stats" parameter. Don't worry about the second part yet, as this can be added later. Currently it would be nice, if always the full information with stats would be returned. But not 100% sure if it is that simple :-)